source/game_sa/CBirds.cpp

```diff
@@ -1 +1,444 @@
+                        break;
+                    case 6:
+                        vecPoint.x = 0.1F;
+                        vecPoint.z = 0.0F;
+                        vertColor = pBird.m_BodyColor;
+                        break;
+                    }
+
+                    vecPoint.y = CBirds::faRenderPosY[uiVertInd];
+                    auto vecWorldPos = matBirdTransform * vecPoint;
+
+                    auto iBufferInd = uiTempBufferVerticesStored + uiVertInd;
+                    auto& pVert1 = aTempBufferVertices[iBufferInd];
+                    pVert1.color = (RwUInt32)CRGBA(vertColor.cRed, vertColor.cGreen, vertColor.cBlue, cAlpha).ToIntARGB();
+                    pVert1.objVertex = vecWorldPos.ToRwV3d();
+                    pVert1.u = CBirds::faRenderCoorsU[uiVertInd];
+                    pVert1.v = CBirds::faRenderCoorsV[uiVertInd];
+
+                    // Mirror on the other side with slightly changed colors
+                    vecPoint.x = -vecPoint.x;
+                    vecWorldPos = matBirdTransform * vecPoint;
+                    vertColor.cRed = static_cast<unsigned char>(static_cast<float>(vertColor.cRed) * 0.8F);
+                    vertColor.cGreen = static_cast<unsigned char>(static_cast<float>(vertColor.cGreen) * 0.8F);
+                    vertColor.cBlue = static_cast<unsigned char>(static_cast<float>(vertColor.cBlue) * 0.8F);
+
+                    auto& pVert2 = aTempBufferVertices[iBufferInd + 8];
+                    pVert2.color = (RwUInt32)CRGBA(vertColor.cRed, vertColor.cGreen, vertColor.cBlue, cAlpha).ToIntARGB();
+                    pVert2.objVertex = vecWorldPos.ToRwV3d();
+                    pVert2.u = CBirds::faRenderCoorsU[uiVertInd];
+                    pVert2.v = CBirds::faRenderCoorsV[uiVertInd];
+                }
+
+                uiTempBufferIndicesStored += 60;
+                uiTempBufferVerticesStored += 16;
+            }
+        }
+        uiUnkn = -1;
+        uiWingMoveTimeOffset += 100;
+    }
+
+    if (uiTempBufferVerticesStored) {
+        RwEngineInstance->dOpenDevice.fpRenderStateSet(rwRENDERSTATESRCBLEND, (void*)5u);
+        RwEngineInstance->dOpenDevice.fpRenderStateSet(rwRENDERSTATEDESTBLEND, (void*)6u);
+        RwEngineInstance->dOpenDevice.fpRenderStateSet(rwRENDERSTATETEXTUREADDRESS, (void*)3u);
+        RwEngineInstance->dOpenDevice.fpRenderStateSet(rwRENDERSTATEFOGENABLE, (void*)0);
+        RwEngineInstance->dOpenDevice.fpRenderStateSet(rwRENDERSTATEVERTEXALPHAENABLE, (void*)1u);
+        RwEngineInstance->dOpenDevice.fpRenderStateSet(rwRENDERSTATETEXTURERASTER, (void*)gpCloudTex[1]->raster);
+        CBrightLights::RenderOutGeometryBuffer();
+    }
+#endif
+}
+
+void CBirds::HandleGunShot(CVector const* pointA, CVector const* pointB)
+{
+#ifdef USE_DEFAULT_FUNCTIONS
+    ((void(__cdecl*)(CVector const*, CVector const*))0x712E40)(pointA, pointB);
+#else
+    auto colLine = CColLine(*pointA, *pointB);
```

Comment:
```suggestion
    CColLine colLine(*pointA, *pointB);
```
This is fine, but I think it would be nice to write it like this.